OrbitQt/servicedeploymanager.cpp

```diff
@@ -0,0 +1,638 @@
+
+  const std::string exe_destination_path = "/tmp/OrbitService";
+  auto& config = std::get<BareExecutableAndRootPasswordDeployment>(
+      deployment_configuration_);
+
+  OUTCOME_TRY(copyFileToRemote(
+      channel, config.path_to_executable.string(), exe_destination_path,
+      OrbitSshQt::SftpOperation::FileMode::UserWritableAllExecutable()));
+
+  OUTCOME_TRY(stopSftpChannel(channel));
+
+  emit statusMessage(
+      "Finished copying the OrbitService executable to the remote instance.");
+  return outcome::success();
+}
+
+outcome::result<void> ServiceDeployManager::startOrbitService() {
+  emit statusMessage("Starting OrbitService on the remote instance...");
+
+  orbit_service_task_.emplace(&session_.value(),
+                              "/opt/developer/tools/OrbitService",
+                              OrbitSshQt::Task::Tty::kNo);
+
+  OrbitSshQt::ScopedConnection quitHandler =
+      QObject::connect(&orbit_service_task_.value(), &OrbitSshQt::Task::started,
+                       &loop, &QEventLoop::quit);
+
+  std::optional<std::error_code> error;
+  OrbitSshQt::ScopedConnection errorHandler = QObject::connect(
+      &orbit_service_task_.value(), &OrbitSshQt::Task::errorOccured, this,
+      [&](std::error_code e) {
+        error = e;
+        loop.exit(2);
+      });
+
+  QObject::connect(
+      &orbit_service_task_.value(), &OrbitSshQt::Task::readyRead, this,
+      [this]() {
+        const auto buf = orbit_service_task_->Read();
+        const auto lines = QString::fromStdString(buf).split('\n');
+        for (const auto& line : lines) {
+          LOG("%s", QString("[OrbitService] %1").arg(line).toStdString());
+        }
+      });
+
+  orbit_service_task_->Start();
+
+  if (error) {
+    return outcome::failure(error.value());
+  }
+
+  switch (const auto result = loop.exec(); result) {
+    case 0:
+      // Successfully started
+      QObject::connect(&orbit_service_task_.value(),
+                       &OrbitSshQt::Task::errorOccured, this,
+                       &ServiceDeployManager::handleSocketError);
+      return outcome::success();
+    case -1:
+      return std::errc::operation_canceled;
+    case 2:
+    default:
+      return outcome::failure(error.value());
+  }
+}
+
+outcome::result<void> ServiceDeployManager::startOrbitServicePrivileged() {
+  emit statusMessage("Starting OrbitService on the remote instance...");
+
+  orbit_service_task_.emplace(&session_.value(), "sudo /tmp/OrbitService",
+                              OrbitSshQt::Task::Tty::kYes);
+
+  const auto& config =
+      std::get<OrbitQt::BareExecutableAndRootPasswordDeployment>(
+          deployment_configuration_);
+
+  OrbitSshQt::ScopedConnection quitHandler = QObject::connect(
+      &orbit_service_task_.value(), &OrbitSshQt::Task::bytesWritten, this,
+      [&, bytes_accumulated = 0](size_t bytesWritten) mutable {
+        bytes_accumulated += bytesWritten;
+        if (bytes_accumulated >= config.root_password.size() + 1) {
```

Comment:
gcc does not compile, because bytes_accumulated is signed int and .size() unsigned aka size_t

Reply:
Thanks for spotting this. :-) I always wanted to push a one character fix once. Fixed.